### PR TITLE
add iam authentication support

### PR DESCRIPTION
**Pull request by sergeyyegournov**
add aws iam authentication support via oauth sasl mechanism

### CODE REVIEW

**Comment by twcurrie** [comment]
#377 looks to cover this, no?

**Comment by twcurrie** [comment]
#372 looks to cover this, no?

**Comment by twcurrie** [comment]
> This PR allows setting the role to assume in providers config as also the aws provider do. For our usecase this would be the better approach.

@Mongey With the above in mind, what would be needed to include these changes as well?

File: kafka/config.go [hunk 1]
```diff
@@ -46,9 +69,12 @@ func (c *Config) newKafkaConfig() (*sarama.Config, error) {
 		case "scram-sha256":
 			kafkaConfig.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient { return &XDGSCRAMClient{HashGeneratorFcn: SHA256} }
 			kafkaConfig.Net.SASL.Mechanism = sarama.SASLMechanism(sarama.SASLTypeSCRAMSHA256)
+		case "iam":
```
Comment (sappusaketh) [inline kafka/config.go:72]: I think this needs to be updated to `aws-iam` as #372 already added this as `aws-iam`

**Comment by sappusaketh** [comment]
@sergeyyegournov can you please update your PR with latest changes on master we do need this feature. If you don't have time please lmk I will open one from my end

**Comment by Mongey** [comment]
superseded by #391

**Pull request closed without merging**